The job **`x86_64-gnu-llvm-12`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/6517889276?check_suite_focus=true) [(plain)](https://github.com/rust-lang/rust/commit/15579d87264ca54cf6ce4e94acd990850423c59e/checks/6517889276/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
..........................................................iiiiii........................ 1144/1199
.......................i...............................
failures:

---- src/ffi/os_str.rs - ffi::os_str::OsString (line 53) stdout ----
error[E0412]: cannot find type `OsStr` in this scope
  |
  |
4 | fn concat_os_strings(a: &OsStr, b: &OsStr) -> OsString {
  |
help: consider importing this struct
  |
3 | use std::ffi::OsStr;
3 | use std::ffi::OsStr;
  |

error[E0412]: cannot find type `OsStr` in this scope
  |
  |
4 | fn concat_os_strings(a: &OsStr, b: &OsStr) -> OsString {
  |
help: consider importing this struct
  |
3 | use std::ffi::OsStr;
3 | use std::ffi::OsStr;
  |

error[E0412]: cannot find type `OsString` in this scope
    |
    |
4   | fn concat_os_strings(a: &OsStr, b: &OsStr) -> OsString {
    |
   ::: /checkout/library/alloc/src/string.rs:366:1
    |
366 | pub struct String {
366 | pub struct String {
    | ----------------- similarly named struct `String` defined here
    |
help: a struct with a similar name exists
    |
4   | fn concat_os_strings(a: &OsStr, b: &OsStr) -> String {
help: consider importing this struct
    |
3   | use std::ffi::OsString;
    |
    |

error[E0433]: failed to resolve: use of undeclared type `OsString`
   --> src/ffi/os_str.rs:55:15
    |
5   |     let ret = OsString::with_capacity(a.len() + b.len()); // This will allocate
    |
   ::: /checkout/library/alloc/src/string.rs:366:1
    |
366 | pub struct String {
366 | pub struct String {
    | ----------------- similarly named struct `String` defined here
    |
help: a struct with a similar name exists
    |
5   |     let ret = String::with_capacity(a.len() + b.len()); // This will allocate
help: consider importing this struct
    |
3   | use std::ffi::OsString;
    |
```

</details>

--- a/library/std/src/ffi/os_str.rs
+++ b/library/std/src/ffi/os_str.rs
@@ -45,10 +45,21 @@ use crate::sys_common::{AsInner, FromInner, IntoInner};
 /// values, encoded in a less-strict variant of UTF-8. This is useful to
 /// understand when handling capacity and length values.
 ///
-/// # Capacity of OsString
+/// # Capacity of `OsString`
 ///
-/// Capacity means UTF-8 byte size for OS strings which created from
-/// valid unicode, and not otherwise specified for other contents.
+/// Capacity uses units of UTF-8 bytes for OS strings which were created from valid unicode, and
+/// uses units of bytes in an unspecified encoding for other contents. On a given target, all
+/// `OsString` and `OsStr` values use the same units for capacity, so the following will work:
+/// ```
+/// use std::ffi::{OsStr, OsString};
+///
+/// fn concat_os_strings(a: &OsStr, b: &OsStr) -> OsString {
+///     let mut ret = OsString::with_capacity(a.len() + b.len()); // This will allocate
+///     ret.push(a); // This will not allocate further
+///     ret.push(b); // This will not allocate further
+///     ret
+/// }
+/// ```
 ///
 /// # Creating an `OsString`
 ///

